### PR TITLE
Added a feature extraction analyzer

### PR DESCRIPTION
**Pull request by kiddinn**
This is the first version of a feature extraction analyzer.

The PR does the following things:
+  Adds the /etc/timesketch folder into the deployment and stores the configuration file for the feature extraction there. 
+ Add a features.conf YAML file where all feature extractions can be defined
+ Adds an analyzer that reads the config file and extracts the features added there
+ Adds a single feature extraction for extracting email addresses from browser history.

### CODE REVIEW

File: timesketch/lib/analyzers/feature_extraction.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        super(FeatureExtractionSketchPlugin, self).__init__(
+            index_name, sketch_id)
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result.
+        """
+        if not os.path.isfile(self.CONFIG_FILE):
```
[the remark this answers is not in the document]
Comment (kiddinn) [reply timesketch/lib/analyzers/feature_extraction.py]: that is true...

File: timesketch/lib/analyzers/feature_extraction.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            return 'Unable to read config file, no features extracted.'
+
+        with open(self.CONFIG_FILE, 'r') as fh:
```
[the remark this answers is not in the document]
Comment (kiddinn) [reply timesketch/lib/analyzers/feature_extraction.py]: agree

File: timesketch/lib/analyzers/interface.py [hunk 3]
```diff
@@ -33,6 +35,27 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+def get_config(file_name):
```
[the remark this answers is not in the document]
Comment (kiddinn) [reply timesketch/lib/analyzers/interface.py]: agree, dome

File: timesketch/lib/analyzers/feature_extraction.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            try:
+                config = yaml.safe_load(fh)
+            except yaml.parser.ParserError as exception:
+                logging.warning((
+                    'Unable to read in YAML config file, '
+                    'with error: {0:s}').format(exception))
+                return 'No results, unable to parse config file.'
+
+        return_strings = []
+        for name, feature_config in config.iteritems():
```
[the remark this answers is not in the document]
Comment (kiddinn) [reply timesketch/lib/analyzers/feature_extraction.py]: yes, I don't think we need that here... not necessarily

File: config/features.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+# The first value extracted is then stored inside the "store_as" attribute.
+# If there are emojis or tags defined they are also applied to that event.
+# In the end, if a view is supposed to be created a view searching for
+# the added tag is added (only if there are results).
+# ------------------------------------------------------------------------
+
+browser_usernames:
+        query_string: 'source_short:"WEBHIST"'
+        attribute: 'message'
+        store_as: 'username'
```
Comment (berggren) [inline config/features.yaml]: Isn't this an email address that is extracted? A username is a bit misleading I think.

Comment (kiddinn) [reply config/features.yaml]: done

**Pull request merged**